clippy_lints/src/if_let_mutex.rs

```diff
@@ -90,12 +91,14 @@ pub struct OppVisitor<'a, 'tcx> {
 }
 
 impl<'tcx> Visitor<'tcx> for OppVisitor<'_, 'tcx> {
-    fn visit_expr(&mut self, expr: &'tcx Expr<'_>) {
+    type Result = ControlFlow<()>;
```

Comment:
This could be `ControlFlow<&'tcx Expr<'tcx>>` with `Break(mutex)`. (Also gets rid of the found_mutex field)